java-squid/src/test/files/se/NullPointerCheck.java

```diff
@@ -354,6 +354,28 @@ public static int testPrimitiveForEachLoopGuava(boolean... values) {
 	    }
 	    return count;
 	  }
+  
+  private  Object a = null;
+  public void newInstance() {
+      if (a == null) {
+        testPrimitiveForEachLoopGood();
+        }
+        a.toString();
+  }
+  protected transient Constructor<?> constructor = null;
+  public void newInstanceBis() throws IllegalAccessException, InstantiationException {
+    try {
+      // Refind the constructor after a deserialization (if needed)
+      if (constructor == null) {
+          this.setDynaBeanClass();
+      }
+      // Invoke the constructor to create a new bean instance
+      return ((DynaBean) constructor.newInstance(constructorValues));
+  } catch (final InvocationTargetException e) {
+      throw new InstantiationException
+              (e.getTargetException().getMessage());
+  }
+}
```

Comment:
This test case seems overcomplicated. All what is not necessary should be removed : (generics, etc).